src/index.js

```diff
@@ -119,10 +119,14 @@ class Dropzone extends React.Component {
     }
   }
 
-  onClick() {
+  onClick(e) {
     if (!this.props.disableClick) {
       this.open();
     }
+
+    if (this.props.onClick) {
+      this.props.onClick.call(this, e);
```

Comment:
I just copied the previous callbacks style, example :https://github.com/okonet/react-dropzone/blob/master/src/index.js#L53-L55

Also not sure if checking is a function is necessary due to it being checked in propTypes?